XenAdmin/Wizards/CrossPoolMigrateWizard/CrossPoolMigrateDestinationPage.cs

```diff
@@ -63,7 +63,8 @@ public CrossPoolMigrateDestinationPage(Host preSelectedHost, List<VM> selectedVM
         public override void PageLoaded(PageLoadedDirection direction)
         {
             base.PageLoaded(direction);
-            PopulateComboBox();
+            if (direction.Equals(PageLoadedDirection.Forward) || ComboBoxEmpty())

```

Comment:
Can we check wether we can do this for ImportSelectHostPage? I noticed it repopulates the combobox too and it may not be necessary now that you have removed the clearing line from the base class's PageLeave method.